File: C/c_00/ex02/ft_print_reverse_alphabet.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   ft_print_reverse_alphabet.c                        :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: photon <marvin@42.fr>                      +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2023/06/21 16:36:20 by photon            #+#    #+#             */
/*   Updated: 2023/06/22 11:32:31 by photon           ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */
#include <unistd.h>

void	ft_print_reverse_alphabet(void)
{
	char	alphabet_char;

	alphabet_char = 'z';
	while (alphabet_char >= 'a')
	{
		write(1, &alphabet_char, 1);
		alphabet_char--;
	}
}
/*
int	main(void)
{
	ft_print_reverse_alphabet();
	return (0);
}
*/

File: zig/zig_00/ex02/ft_print_reverse_alphabet.zig
const std = @import("std");

pub fn ft_putchar(c: u8) void {
    const buf: [1]u8 = [_]u8{c};
    _ = std.os.write(1, &buf) catch return;
}

pub fn ft_print_reverse_alphabet() void {
    var buf: u8 = 'z';
    while (buf >= 'a')
    {
        ft_putchar(buf);
        buf -= 1;
    }
}

pub fn main() !void {
    ft_print_reverse_alphabet();
}

